doc/source/development/rfc/rfc100_float16_support.rst

```diff
@@ -0,0 +1,132 @@
+.. _rfc-100:
+
+=============================
+RFC 100: Support float16 type
+=============================
+
+============== =============================================
+Author:        Erik Schnetter
+Contact:       eschnetter @ perimeterinstitute.ca
+Started:       2024-Jun-05
+Status:        Proposed
+Target:        GDAL 3.10
+============== =============================================
+
+Summary
+-------
+
+This RFC adds support for the IEEE 16-bit floating point data type
+(aka ``half``, ``float16``). It adds a new pixel data type
+``GDT_Float16``. Where supported by the compiler, it also adds a new
+C/C++ type alias ``GFloat16``. Even when the local system does not
```

Comment:
The Gxxxx typedefs are mostly of legacy interest. They were introduced long before C/C++ standardized fixed-size types ([u]int[8/16/32/64] etc), and we hope to get rid of them at some point (cf https://gdal.org/development/rfc/rfc95_standard_int_types.html , which is postponed to an hypothetical GDAL 4.0). So here I would go to just using ``_float16`` (is that standardized for C ? I saw that C++23 has standardized std::float16, but do all C++ implementations that understand std::float16 understand ``_float16`` ?)